File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        ans,i,j=0,0,len(height)-1
        while i<j:
            contain=(j-i)*min(height[i],height[j])
            ans=max(ans,contain)
            if height[i]<height[j]:i+=1
            elif height[j]<height[i]:j-=1
            else: i+=1
        return ans

File: 0011-container-with-most-water/0011-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& h) {
        int maxarea=0,l=0,r=h.size()-1;
        while(l<r){
            int area=(r-l)*min(h[l],h[r]);
            if(h[r]>h[l]) l++;
            else r--;
            if(area>maxarea) maxarea=area;
        }
        return maxarea;
    }
};
